File: javaScript/problem3.js
var factor = [];
var i = 2;
var prime = function(number) {
    while (i * i <= number) {        
        while (number % i === 0) {
            number = number / i;
            
        }
        i += 1;
    }
    return number;
};
prime(600851475143);

File: ruby/problem3.rb
# Solution taken from dstodolny (https://github.com/dstodolny/project_euler/blob/master/problem003.rb)

def factor_finder (number, factor = 2)
	if number >= factor
		(number % factor == 0) ? factor_finder(number/factor, factor) || factor : factor_finder(number, factor + 1)
	end
end

puts factor_finder(600851475143)